### PR TITLE
LOG-597. Impl IndexManagement Spec

### PR DESCRIPTION
**Pull request by jcantrill**
Implement the IndexManagementSpec

Ref https://jira.coreos.com/browse/LOG-597

### CODE REVIEW

File: pkg/apis/logging/v1/elasticsearch_types.go [hunk 1]
```diff
@@ -48,6 +48,7 @@ type ElasticsearchSpec struct {
 	RedundancyPolicy RedundancyPolicyType  `json:"redundancyPolicy"`
 	Nodes            []ElasticsearchNode   `json:"nodes"`
 	Spec             ElasticsearchNodeSpec `json:"nodeSpec"`
+	IndexManagement  *IndexManagementSpec  `json:"indexManagement"`
```
Comment (ewolinetz) [inline pkg/apis/logging/v1/elasticsearch_types.go]: do we want this to be able to be `nil`?

File: pkg/apis/logging/v1/elasticsearch_types.go [hunk 2]
```diff
@@ -63,6 +64,7 @@ type ElasticsearchStatus struct {
 	ShardAllocationEnabled ShardAllocationState                  `json:"shardAllocationEnabled"`
 	Pods                   map[ElasticsearchNodeRole]PodStateMap `json:"pods"`
 	Conditions             []ClusterCondition                    `json:"conditions"`
+	IndexManagementStatus  *IndexManagementStatus                `json:"indexManagement,omitempty"`
```
Comment (ewolinetz) [inline pkg/apis/logging/v1/elasticsearch_types.go]: same ask here

File: pkg/apis/logging/v1/elasticsearch_types.go (hunk 1 above)
Comment (jcantrill) [reply pkg/apis/logging/v1/elasticsearch_types.go]: Yes, represents undefined.  I would assume Jaeger would like to move to this impl of IndexManagement but they may not.

**Pull request merged**